### ISSUE TITLE
Runtime errors when fuzzing a public function taking string-ascii

### ISSUE THREAD

**Issue by moodmosaic** [hirosystems/clarinet#496]
### Contract

```clarity
(define-data-var str (string-ascii 256) "")

(define-public (set-str (new-str (string-ascii 256)))
    (begin
        (ok (var-set str new-str))
    )
)
```

### Fuzz

```ts
import {
  Account,
  Chain,
  Clarinet,
  Tx,
  types,
} from "https://deno.land/x/clarinet@v0.31.1/index.ts";

// FIXME https://github.com/hirosystems/clarinet/issues/393
// FIXME import fc from "https://esm.sh/fast-check";
import fc from "https://cdn.skypack.dev/fast-check"; // Workaround for #393.

Clarinet.test({
  name: "[Exampe] ascii fuzz-test",
  fn(chain: Chain, accounts: Map<string, Account>) {
    fc.assert(
      fc.property(
        fc.ascii(), // https://ss64.com/ascii.html
        fc.constantFrom(...accounts.values()),
        (str: string, sender: Account) => {
          console.log(str);

          const block = chain.mineBlock([
            Tx.contractCall(
              "example",
              "set-str",
              [types.ascii(str)],
              sender.address,
            ),
          ]);
          console.log(block);
        },
      ),
      { numRuns: 10, unbiased: false }, // https://github.com/dubzzz/fast-check/discussions/2978
    );
  },
});
```

### Output

```
$ clarinet --version
clarinet 0.32.0

$ clarinet test
Running file:///.../empty-project/tests/example_test.ts
>
{ height: 2, receipts: [ { result: "(ok true)", events: [] } ] }
▲
Runtime error: example::set-str("▲") -> illegal ascii string "▲"
{ height: 3, receipts: [] }
▼
Runtime error: example::set-str("▼") -> use of unresolved contract 'ST2CY5V39NHDPWSXMW9QDT3HC3GD6Q6XX4CFRK9AG.example'
{ height: 4, receipts: [] }
⌂
Runtime error: example::set-str("⌂") -> use of unresolved contract 'STNHKEPYEPJ8ET55ZZ0M5A34J0R3N5FM2CMMMAZ6.example'
{ height: 5, receipts: [] }
&
Runtime error: example::set-str("&") -> use of unresolved contract 'STNHKEPYEPJ8ET55ZZ0M5A34J0R3N5FM2CMMMAZ6.example'
{ height: 6, receipts: [] }
`
{ height: 7, receipts: [ { result: "(ok true)", events: [] } ] }
t
{ height: 8, receipts: [ { result: "(ok true)", events: [] } ] }
&
{ height: 9, receipts: [ { result: "(ok true)", events: [] } ] }
#
{ height: 10, receipts: [ { result: "(ok true)", events: [] } ] }
►
Runtime error: example::set-str("►") -> illegal ascii string "►"
{ height: 11, receipts: [] }
* [Exampe] ascii fuzz-test ... ok (111ms)
```

### Notes

1. What all these `use of unresolved contract 'ST...` mean? At the very least, I would expect to get only `illegal ascii string "..."`.
2. It looks as the test actually passes (`* [Example] ascii fuzz-test ... ok (90ms)`) even though it (partially?) failed on runtime. This looks like a [false negative](http://xunitpatterns.com/false%20negative.html) to me.

/cc @jcnelson

**Comment by whoabuddy** [comment]
Also tagging @obycode and @lgalabru on this one, in the past when I've run into these `use of unresolved contract` errors it was usually when I was using multiple contracts and my test script caused one of them to panic for some reason, and the contract wouldn't deploy.

I'm surprised the tests continued given the runtime errors, iirc mine would stop at where the error was generated and we'd have to turn around and figure out what happened, so I find it odd that it continued in the first place. The final output seems to be missing some data as well.

PS - love the fuzzing techniques here! Those tests be like...

![image](https://user-images.githubusercontent.com/9038904/180305893-478d6790-17fd-4ca1-906b-6183da7242ae.png)

**Comment by moodmosaic** [comment]
>it was usually when I was using multiple contracts

Interesting.... In this case, only 1 contract was used. In fact I shrinked `Clarinet.toml` to be like

```
$ cat Clarinet.toml

[project]
name = "empty-project"
telemetry = false

[contracts.example]
path = "contracts/example.clar"
```

**Comment by obycode** [comment]
I'm looking into the "use of unresolved contract" errors, but I don't think it is a false pass -- the test is just missing the necessary checks. I would add a check that `block.receipts.length` is equal to 1.

**Comment by moodmosaic** [comment]
The test runs 10 times (configurable). Asserting that block height is always 1 will cause all other tests to fail, no?

**Comment by obycode** [comment]
No, that's not checking the block height, just the count of the receipts in the single block.

**Comment by obycode** [comment]
It looks like a string with a `"` is not being handled correctly somewhere in the testing framework.

**Comment by obycode** [comment]
```ts
  export function ascii(val: string) {
    return `"${val}"`;
  }

  export function utf8(val: string) {
    return `u"${val}"`;
  }
```

These need to escape `"` characters. Otherwise, `"""` is passed as the string parameter.

**Comment by obycode** [comment]
There also seems to be some error with the testing framework in the handling of runtime errors. I'll dig further into that too.

**Comment by moodmosaic** [comment]
> No, that's not checking the block height, just the count of the receipts in the single block.

Oh, you're right. I should read more carefully.

**Comment by obycode** [comment]
Found the problem causing the unresolved contracts. I'll open a PR shortly for both issues.

**Comment by moodmosaic** [comment]
Thanks @obycode!

Let's fuzz all the things and find bugs for free.

**Comment by moodmosaic** [comment]
Resolved by #500. If not, we can reopen. Thanks @obycode :+1:
